music21/interval.py

```diff
@@ -3490,10 +3493,40 @@ def _diatonicTransposePitch(self,
                 # cannot import Accidental here.
                 pitch2.accidental = halfStepsToFix  # type:ignore
 
-            # inherit accidental display type etc. but not current status
-            if pitch2.accidental is not None and pitch1.accidental is not None:
-                pitch2.accidental.inheritDisplay(pitch1.accidental)
-                pitch2.accidental.displayStatus = None  # set accidental display to None
+            # inherit accidental display options
```

Comment:
I'm not sure what is happening here that wasn't covered in the previous lines -- this whole set of branches is confusing to me, and there are several possibilities that we're creating accidentals but possibly not using them (which can lead up to a 20% performance slow-down).  What's the particular case that was broken before but works now?

Reply:
Rewritten.